### PR TITLE
Restore coverage

### PR DESCRIPTION
**Pull request by yyoncho**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -140,14 +140,14 @@ jobs:
...
-      #       ~/.nimble/packages_official.json
-      #     key: ${{ inputs.os }}-${{ inputs.cpu }}-${{ env.cache_nonce }}
+      - name: Restore nimble dependencies from cache
+        id: nimble_deps
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.nimble/pkgs2
+            ~/.nimble/packages_official.json
+          key: ${{ inputs.os }}-${{ inputs.cpu }}-${{ env.cache_nonce }}
```
Comment (elcritch) [inline .github/workflows/ci.yml:150]: I'd re-added this piece on main. Though I just did `~/.nimble` which seems to work.

File: .github/workflows/codecov.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+          fetch-depth: 0
+
+      - name: Environment setup
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y lcov build-essential git curl
+          mkdir coverage
+
+      - name: Setup Nimble
+        uses: "./.github/actions/install_nimble"
```
Comment (elcritch) [inline .github/workflows/codecov.yml:35]: Is this something we could make into a global github action? Something like https://github.com/jiro4989/setup-nim-action

Comment (yyoncho) [reply .github/workflows/codecov.yml:35]: https://github.com/nim-lang/nimble/pull/1122

**Review by dryajov — APPROVED** [review]

**Comment by elcritch** [comment]
Ok, this is building. Gonna merge since it's working and restores previous CI feature.

**Pull request merged**